### PR TITLE
Rename `Snapshot` to `Frame`.

### PR DESCRIPTION
**Pull request by joaander**
## Description

* Rename `Snapshot` to `Frame` and deprecate `Snapshot`.
* Continue providing `Snapshot` for backwards compatibility in v2.

## Motivation and Context

Users confuse `gsd.hoomd.Snapshot` with `hoomd.Snapshot`. Rename the former to `gsd.hoomd.Frame` to prevent confusion.

## How Has This Been Tested?

I reviewed the documentation and updated all unit tests. For compatibility with `isinstance` checks, `Frame` is a subclass of `Snapshot`, so all unit tests implicitly test `Snapshot` and `Frame`.

## Change log

```
Added:

* ``gsd.hoomd.Frame`` class to replace ``gsd.hoomd.Snapshot``.

Deprecated:

* ``gsd.hoomd.Snapshot``.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/trunk-patch/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/trunk-patch/ContributorAgreement.md).
- [x] My name is on the list of contributors (`doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: gsd/test/test_fl.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def test_create(tmp_path):
 ])
 def test_dtype(tmp_path, typ):
     """Test all supported data types."""
-    data1d = numpy.array([1, 2, 3, 4, 5, 10012], dtype=typ)
+    data1d = numpy.array([1, 2, 3, 4, 5, 127], dtype=typ)
```
Comment (tommy-waltmann) [inline gsd/test/test_fl.py:43]: Why this change?

**Review by tommy-waltmann — COMMENTED** [review]
These changes seem like a good idea. Is there a proposed timeline for the gsd version 3 release?

File: gsd/test/test_fl.py (hunk 1 above)
Comment (joaander) [reply gsd/test/test_fl.py:43]: This quiets a deprecation warning from numpy. `test_dtype` is called with the `int8` data type which cannot represent the value 10012.

**Comment by joaander** [comment]
I do not have a particular schedule for gsd 3.0 at this time.

**Review by tommy-waltmann — APPROVED** [review]
